app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/TravelCertificateHistory.kt

```diff
@@ -0,0 +1,422 @@
+      viewModel.emit(CertificateHistoryEvent.DownloadCertificate(url))
+    },
+    onContinue = onContinue,
+    navigateUp = navigateUp,
+    onShareTravelCertificate = onShareTravelCertificate,
+    uiState = uiState,
+  )
+}
+
+@Composable
+private fun TravelCertificateHistoryScreen(
+  reload: () -> Unit,
+  onCertificateClick: (String) -> Unit,
+  onContinue: () -> Unit,
+  navigateUp: () -> Unit,
+  onShareTravelCertificate: (TravelCertificateUri) -> Unit,
+  uiState: CertificateHistoryUiState,
+) {
+  var showBottomSheet by remember { mutableStateOf(false) }
+  val explanationSheetState = rememberModalBottomSheetState(true)
+  if (showBottomSheet) {
+    TravelCertificateInfoBottomSheet(
+      onDismiss = { showBottomSheet = false },
+      sheetState = explanationSheetState,
+    )
+  }
+
+  when (uiState) {
+    CertificateHistoryUiState.FailureDownloadingCertificate -> {
+      ErrorDialog(
+        title = stringResource(id = R.string.general_error),
+        message = stringResource(id = R.string.travel_certificate_downloading_error),
+        onDismiss = {
+          reload()
+        },
+      )
+    }
+
+    CertificateHistoryUiState.FailureDownloadingHistory -> {
+      HedvigScaffold(
+        navigateUp = navigateUp,
+        modifier = Modifier.clearFocusOnTap(),
+        topAppBarText = stringResource(id = R.string.PROFILE_ROW_TRAVEL_CERTIFICATE),
+        topAppBarActions = {
+          IconButton(
+            onClick = { showBottomSheet = true },
+            modifier = Modifier.size(40.dp),
+          ) {
+            Icon(
+              imageVector = Icons.Hedvig.Info,
+              contentDescription = stringResource(R.string.REFERRALS_INFO_BUTTON_CONTENT_DESCRIPTION),
+              modifier = Modifier.size(24.dp),
+            )
+          }
+        },
+      ) {
+        Spacer(modifier = Modifier.weight(1f))
+        HedvigInformationSection(
+          title = stringResource(id = R.string.something_went_wrong),
+          buttonText = stringResource(id = R.string.GENERAL_RETRY),
+          onButtonClick = {
+            reload()
+          },
+        )
+        Spacer(modifier = Modifier.weight(1f))
+      }
+    }
+
+    CertificateHistoryUiState.Loading -> {
+      Box(modifier = Modifier.fillMaxSize()) {
+        CircularProgressIndicator(modifier = Modifier.align(Alignment.Center))
+      }
+    }
+
+    is CertificateHistoryUiState.SuccessDownloadingCertificate -> {
+      LaunchedEffect(uiState.downLoadingUri) {
+        uiState.downLoadingUri?.let {
+          onShareTravelCertificate(it)
+        }
+      }
+    }
+
+    is CertificateHistoryUiState.SuccessDownloadingHistory -> {
+      ShowHistory(
+        onIconClick = { showBottomSheet = true },
+        onCertificateClick = onCertificateClick,
+        onContinue = onContinue,
+        navigateUp = navigateUp,
+        historyList = uiState.certificateHistoryList,
+      )
+    }
+  }
+}
+
+@Composable
+private fun ShowHistory(
```

Comment:
This composable being a verb can be a tiny bit confusing when looked at from the outside. How about `TravelCertificateSuccessScreen`?